app/models/financialDetails/FinancialDetail.scala

```diff
@@ -88,11 +89,16 @@ case class FinancialDetail(taxYear: String,
     }
     .filter(_.getPaymentAllocationTextInChargeSummary.isDefined)
 
-  def isMFACredit: Boolean = MfaCreditUtils.validMFACreditType(mainType)
-
-  def isCutOverCredit: Boolean = mainType.contains("ITSA Cutover Credits")
-
-  def isBalancingChargeCredit: Boolean = mainType.contains("SA Balancing Charge Credit")
+  def getCreditType: Option[CreditType] = {
+    val validMFA = MfaCreditUtils.validMFACreditType(mainType)
+    (validMFA, mainType) match {
+      case (true, _) => Some(MfaCreditType)
+      case (_, Some(credit)) => if (credit.contains("ITSA Cutover Credits")) Some(CutOverCreditType) else {
```

Comment:
you re-write
`case (_, Some(credit)) => if (credit.contains("ITSA Cutover Credits")) Some(CutOverCreditType)`
into:
`case (_, Some(""ITSA Cutover Credits"")) => Some(CutOverCreditType)`

Reply:
I wasn't sure if the mainType would be exactly "ITSA Cutover Credits" or whether it could have extra text and so we would need to do .contains. But if it will always be "ITSA Cutover Credits" and never "...ITSA Cutover Credits..." I will change it

Reply:
you can check ATs project for exact data would be in this field; my assumption is that this would be exact match, but you can apply something like: ~.map(_.trim) to drop any extra space; but this is very unlikely

Reply:
Where in the ATs would I found that?